pandas/io/formats/format.py

```diff
@@ -1647,10 +1657,26 @@ def get_format_datetime64(
     a string as output"""
 
     if is_dates_only:
+        str_date_fmt = loc_s = None
+        if date_format is not None:
+            try:
+                # Try to get the string formatting template for this format
+                str_date_fmt, loc_s = convert_strftime_format(
+                    date_format, target="datetime"
+                )
+            except UnsupportedStrFmtDirective:
+                # Unsupported directive: fallback to standard `strftime`
+                pass
         return lambda x: _format_datetime64_dateonly(
-            x, nat_rep=nat_rep, date_format=date_format
+            x,
+            nat_rep=nat_rep,
+            date_format=date_format,
+            str_date_fmt=str_date_fmt,
+            loc_s=loc_s,
         )
     else:
+        # Relies on datetime.str, which is fast already
+        # TODO why is date_format not used in this case ? This seems like a bug?
```

Comment:
The only place where this function is used with a non-None date_format seems to be within the `_Datetime64TZFormatter` class. 

I do not clearly understand from the code base the purpose of this class, as it is not used in the CSV formatter for example. Maybe it is deprecated ? Anyway, the type hint for the `date_format` attribute is `None` suprisingly, so it explains why not using it is not causing much trouble : https://github.com/pandas-dev/pandas/blob/70fc17444be82a8e07bfc1b69088379b79a3653b/pandas/io/formats/format.py#L1493